vmdb/app/models/mixins/retirement_mixin.rb

```diff
@@ -188,4 +187,18 @@ def retiring?
   def error_retiring?
     retirement_state == ERROR_RETIRING
   end
+
+  private
+
+  def setup_event_hash(requester)
+    event_hash = {:retirement_initiator => "system"}
+    event_hash[retirement_base_model_name.underscore.to_sym] = self
+    event_hash[:host] = host if self.respond_to?(:host)
+    if requester
+      event_hash[:user_id] = requester
+      event_hash[:retirement_initiator] = "user"
+    end
+    event_hash[:type] ||= self.class.name
+    event_hash
```

Comment:
Does this event_hash end up on the queue?  If so, won't those AR objects be huge?